var/spack/repos/builtin/packages/crtm/package.py

```diff
@@ -13,13 +13,58 @@ class Crtm(CMakePackage):
     scattering, and a solver for a radiative transfer."""
 
     homepage = "https://www.jcsda.org/jcsda-project-community-radiative-transfer-model"
-    url = "https://github.com/NOAA-EMC/EMC_crtm/archive/refs/tags/v2.3.0.tar.gz"
+    git = "https://github.com/JCSDA/crtm.git"
+    url = "https://github.com/JCSDA/crtm/archive/refs/tags/v2.3.0.tar.gz"
 
     maintainers = [
+        "BenjaminTJohnson",
         "t-brown",
         "edwardhartnett",
         "AlexanderRichert-NOAA",
         "Hang-Lei-NOAA",
+        "climbfuji",
     ]
 
-    version("2.3.0", sha256="3e2c87ae5498c33dd98f9ede5c39e33ee7f298c7317b12adeb552e3a572700ce")
+    variant(
+        "fix",
+        default=False,
+        description='Download CRTM coeffecient or "fix" files (several GBs).',
+    )
+    variant(
+        "build_type",
+        default="RelWithDebInfo",
+        description="CMake build type",
+        values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel"),
+    )
+
+    depends_on("cmake@3.15:")
+    depends_on("git-lfs")
+    depends_on("netcdf-fortran", when="@2.4.0:")
+    depends_on("netcdf-fortran", when="@v2.3-jedi.4")
+    depends_on("netcdf-fortran", when="@v2.4-jedi")
+    depends_on("netcdf-fortran", when="@v2.4-jedi.1")
+    depends_on("netcdf-fortran", when="@v2.4-jedi.2")
+
+    depends_on("crtm-fix@2.3.0_emc", when="@2.3.0 +fix")
+    depends_on("crtm-fix@2.4.0_emc", when="@2.4.0 +fix")
+
+    depends_on("ecbuild", type=("build"), when="@v2.3-jedi.4")
+    depends_on("ecbuild", type=("build"), when="@v2.4-jedi")
+    depends_on("ecbuild", type=("build"), when="@v2.4-jedi.1")
+    depends_on("ecbuild", type=("build"), when="@v2.4-jedi.2")
+
+    # ecbuild release v2.4.0 is broken
+    # add ecbuild dependency for next release with fix
+    # depends_on("ecbuild", when="@2.4.0:", type=("build"))
+
+    # REL-2.4.0_emc (v2.4.0 ecbuild does not work)
+    version("2.4.0", commit="5ddd0d6")
+    # Uses the tip of REL-2.3.0_emc branch
+    version("2.3.0", commit="99760e6")
+    # JEDI applications so far use these versions
+    # Branch release/crtm_jedi
+    version("v2.3-jedi.4", commit="bfede42")
+    # Branch release/crtm_jedi_v2.4.0
+    version("v2.4-jedi", commit="0ee3593")
+    version("v2.4-jedi.1", commit="8222341")
+    version("v2.4-jedi.2", commit="62831cb")
```

Comment:
As far as I can tell, spack does not distinguish between "v2.4-jedi" and "v2.4-jedi.#" in the sense that when I add all these to my environment, "crtm@v2.4-jedi" gets concretized to, e.g., "crtm@v2.4-jedi.4".

Reply:
I'll need to check, so far I always got what  I wanted (but I never asked for `crtm@v2.4-jedi` and `crtm@v2.4-jedi.2`, only for `crtm@v2.4-jedi.1` and `crtm@v2.4-jedi.2`

Reply:
I cannot reproduce the problem, also not for `2.4.0` vs `2.4.0_emc` etc. I am attaching the my skylab-3.0.0 environment (coming from our `skylab-3.0.0` template) and the log files for install/concretization, I just built this today. It's clearly adding `2.4.0`, `2.3.0`, `crtm@v2.3-jedi.4` and `crtm@v2.4-jedi.2`, just as requested.

[skylab-3.0.0-env.tar.gz](https://github.com/spack/spack/files/10317494/skylab-3.0.0-env.tar.gz)

Reply:
Oh, I of course did not install the `+fix` variant, therefore that's something we'll need to check for an environment that actually uses it (`hpc-dev-v1`).

Reply:
Ok, I _was_ able to reproduce the problem for `crtm@v2.4-jedi`. The best solution is to simply remove this version, because `.1` and `.2` are bug fix releases for the same code. Nobody will want to install the earlier version with known bugs. Thanks for pointing this out.

As for the `crtm-fix` package, the EMC versions of crtm `2.3.0` and `2.4.0` ask for `crtm-fix` versions `2.3.0_emc` and `2.4.0_emc`, no configuration wants `crtm-fix@2.4.0`. So I suggest to remove that version as well, but it would be good to hear from you or other EMC folks and from @BenjaminTJohnson if that is ok.

Reply:
I see no issue with this.  

Also, I'm completely open to recommendations for spack friendly / best practices for version tag names.   This was a decision that was intended to follow semantic versioning (https://semver.org/).